### DISCUSSION TITLE
vector brush with fill tool

### DISCUSSION THREAD

**Discussion by trinhpxl** [keiwando/toonsquid#40] [Q&A] [answered]
hihi, someone i knew was having trouble filling in color within vector brush lines so i tried to make a project helping them, but then noticed the fill tool was filling weird ? 

i also had this same issue with a another project of mine before where i did lineart in vector and tried filling with color, and it just never worked ? so i just redid the lineart in raster lmao— is this how its supposed to work or am i doing something wrong ?

https://user-images.githubusercontent.com/115050959/198899442-6cfd6ad3-fca3-462a-9ba2-6b828fb55cc8.MOV

**Comment by keiwando** [comment]
Yeah, the vector fill tool can unfortunately be a bit finicky at times, especially when the surrounding region is made of a lot of different lines. What you see there is definitely not how it's supposed to work, but I won't go into the math-heavy details for what exactly is happening since that shouldn't matter to anyone using the tool.

I'll definitely be working on trying to make the vector fill tool a lot more reliable than it currently is, but since it works in a very different and much more complicated way than the pixel fill tool there's unfortunately no quick fix for these scenarios right now. Sorry about that.

If you happen to run into more cases like this, you could also upload the project file here so that I can have more reference for exact situations in which the vector fill tool currently fails. (Feel free to duplicate it and delete everything else except for the drawing / layers that are causing issues to make the project easier to share)

**Comment by keiwando** [comment, marked as answer]
I have implemented some reliability improvements for the vector fill tool that are now released with the 1.7.1 update. These cases can still occur, but should now be a lot less likely. I will also still continue to work on this over the course of other updates in the future.
